### DISCUSSION TITLE
Hooking a plugin manager into lz.n

### DISCUSSION THREAD

**Discussion by IogaMaster** [nvim-neorocks/lz.n#46] [Q&A] [answered]
Can a custom handler be used to install plugins using mini.deps

**Comment by IogaMaster** [comment]
Or is it more of a custom function that uses grabs the info from the plugin spec and then calls install

**Comment by mrcjkb** [comment, marked as answer]
I don't think a custom handler can be used to install plugins. It's meant for loading them.
But it should be possible to use lz.n to lazy-load plugins that are managed with mini.deps.
I haven't tried it though.

**Comment by IogaMaster** [comment]
Mini.deps loads at the same time as the install, so looks like i will have to make my own plugin manager that uses lz.n under the hood

**Reply by IogaMaster** [reply]
Actually, I can use the before attribute to call the install

**Reply by mrcjkb** [reply]
You could also use a `table<string, function>` where the key is the plugin name and the value is a function that loads/installs the plugin with mini.deps.
Then you could set the `load` attribute to a function that performs a lookup.

But maybe using the `before` hook is simpler in this case.
